homeassistant/components/mysensors/const.py

```diff
@@ -1,19 +1,33 @@
 """MySensors constants."""
 from collections import defaultdict
-from typing import Dict, List, Set, Tuple
+from typing import Dict, List, Literal, Set, Tuple
 
 ATTR_DEVICES: str = "devices"
 
 CONF_BAUD_RATE: str = "baud_rate"
 CONF_DEVICE: str = "device"
 CONF_GATEWAYS: str = "gateways"
 CONF_NODES: str = "nodes"
+CONF_PERSISTENCE: str = "persistence"
 CONF_PERSISTENCE_FILE: str = "persistence_file"
 CONF_RETAIN: str = "retain"
 CONF_TCP_PORT: str = "tcp_port"
 CONF_TOPIC_IN_PREFIX: str = "topic_in_prefix"
 CONF_TOPIC_OUT_PREFIX: str = "topic_out_prefix"
 CONF_VERSION: str = "version"
+CONF_GATEWAY_TYPE: str = "gateway_type"
+CONF_GATEWAY_TYPE_TYPE = Literal[
+    "gateway_type_serial", "gateway_type_ethernet", "gateway_type_mqtt"
```

Comment:
Technically we only support the TCP gateway type. MySensors have an option for UDP which we don't support for ethernet gateway.